src/columns.jl

```diff
@@ -186,8 +186,17 @@ Base.IndexStyle(::Type{<:Columns}) = IndexLinear()
 summary(c::Columns{D}) where {D<:Tuple} = "$(length(c))-element Columns{$D}"
 
 empty!(c::Columns) = (foreach(empty!, c.columns); c)
-similar(c::Columns{D,C}) where {D,C} = Columns{D,C}(map(similar, c.columns))
-similar(c::Columns{D,C}, n::Integer) where {D,C} = Columns{D,C}(map(a->similar(a,n), c.columns))
+
+function similar(c::Columns{D,C}) where {D,C}
+    cols = map(similar, c.columns)
+    Columns{D,typeof(cols)}(cols)
```

Comment:
How bad is it to lessen the inferrability of the output?

Reply:
This call should be inferrable. :+1: